src/dbnode/network/server/tchannelthrift/node/service.go

```diff
@@ -1298,23 +1305,23 @@ func (r *writeBatchPooledReq) Finalize() {
 	}
 	r.pooledIDsUsed = 0
 
-	// Return any pooled thrift byte slices to the thrift pool
+	// Return any pooled thrift byte slices to the thrift pool.
 	if r.writeReq != nil {
 		for _, elem := range r.writeReq.Elements {
 			apachethrift.BytesPoolPut(elem.ID)
-			if elem.Datapoint.Annotation != nil {
-				apachethrift.BytesPoolPut(elem.Datapoint.Annotation)
-			}
+			// Don't return the annotations to the pool because they don't
```

Comment:
the first sentence in the comment is confusing unless people understand nuances of when IDs are cloned v when retained. much simpler to just rely on the second sentence and say the ownership of the lifecycle for the annotations has been transferred to the `BatchWriter`.

that said, can we do the same thing for the ID/EncodedTags too? i.e. expect the BatchWriter to clean it up? might cause a minor rise in steady state memory usage but would simplify this code a lot.

Reply:
Fixed the comment. Spoke offline and decided not to do the ID/EncodedTags thing for performance reasons (more synchronization required by single-threaded commitlog)